recurring_contract/model/contract_group.py

```diff
@@ -82,22 +91,102 @@ def _get_groups_from_contract(self, cr, uid, ids, context=None):
                     _get_groups_from_contract, ['next_invoice_date',
                                                 'state'], 20),
             }),
+        'last_paid_invoice_date': fields.function(
+            _get_last_paid_invoice, type='date',
+            string=_('Last paid invoice date')
+        )
     }
 
     _defaults = {
         'ref': '/',
         'recurring_unit': 'month',
         'recurring_value': 1,
+        'advance_billing_months': 1,
     }
 
+    def _get_contract_ids(self, cr, uid, ids, context=None):
+        contract_ids = list()
+
+        for contract_group_id in self.browse(cr, uid, ids, context):
+            for contract in contract_group_id.contract_ids:
+                if contract.id not in contract_ids:
+                    contract_ids.append(contract.id)
+        return contract_ids
+
     def write(self, cr, uid, ids, vals, context=None):
+
+        recurring_contract_obj = self.pool.get('recurring.contract')
+        contract_ids = self._get_contract_ids(cr, uid, ids, context)
+
+        if ('advance_billing_months' in vals):
+            old_advance_billing_months = self.browse(
+                cr, uid, ids, context)[0].advance_billing_months
+            if old_advance_billing_months > vals['advance_billing_months']:
+                pdb.set_trace()
+                self._on_advance_billing_changed(
+                    cr, uid, contract_ids,
+                    vals['advance_billing_months'],
+                    context)
+                since_date = datetime.today() + \
+                    relativedelta(months=+vals['advance_billing_months'])
+                recurring_contract_obj.clean_invoices(
+                    cr, uid, contract_ids, context=context,
+                    since_date=since_date)
+
+            self.button_generate_invoices(cr, uid, ids, context=context)
+
+        if ('recurring_value' in vals or
+                'recurring_unit' in vals):
+            self.button_generate_invoices(cr, uid, ids, context=context)
+
         res = super(contract_group, self).write(cr, uid, ids, vals, context)
-        if 'advance_billing' in vals:
-            self.generate_invoices(cr, uid, ids, context=context)
         return res
 
+    def _on_next_invoice_change(

```

Comment:
Method is never called (don't know if it's in progress but I still write it in case).